### PR TITLE
Calendar sync with phone native calendar

### PR DESCRIPTION
**Pull request by Grisou13**

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -29,7 +29,9 @@
 <body>
 
   <!-- Ionic's root component and where the app will load -->
+  
   <ion-app></ion-app>
+  <p>prout</p>
```
Comment (kayoumido) [inline src/index.html]: Like come one ....

File: src/pages/calendar/calendar.ts [hunk 2]
```diff
@@ -31,11 +33,24 @@ export class CalendarPage {
...
+    }))
+  }
+  @ViewChild("sync-btn") syncBtn;
   @ViewChild('calendar') calendar;
-  constructor(public navCtrl: NavController, private loadingCtrl: LoadingController, public navParams: NavParams, public InternetStatus : InternetStatusProvider, private userService : UserService) {
+  constructor(public navCtrl: NavController, private internetStatusProvider : InternetStatusProvider, private toastCtrl: ToastController, private loadingCtrl: LoadingController, private alertCtrl : AlertController, public navParams: NavParams, public InternetStatus : InternetStatusProvider, private userService : UserService, private calendarService: Calendar) {
   }
 
   ionViewWillLoad() {
+    this.calendarService.listEventsInRange(new Date("01-02-2018"), new Date()).then(console.log).catch(console.log);
```
Comment (kayoumido) [inline src/pages/calendar/calendar.ts]: Does the date need to be hard coded ?

File: src/pages/calendar/calendar.ts [hunk 3]
```diff
@@ -63,15 +79,7 @@ export class CalendarPage {
 
   loadCalendar(){
     console.log(this)
```
Comment (kayoumido) [inline src/pages/calendar/calendar.ts]: Is this really needed ?

File: src/pages/calendar/calendar.ts [hunk 4]
```diff
@@ -96,5 +104,73 @@ export class CalendarPage {
...
+    this.calendarService.hasReadWritePermission()
+      .then(has => has ? true : this.calendarService.requestReadWritePermission())
+      .then(()=>this.calendarService.deleteCalendar(calendarOptions.calendarName))
+      .then(()=>console.debug("deleted calendar successfully"))
+      .then(()=>this.calendarService.createCalendar(calendarOptions))
+      .then(()=>console.debug("created calendar successfully"))
+      .then(()=>this.calendarService.listCalendars())
+      .then(calendarList => {
+        //build calendar options
+        console.log(calendarList)
```
Comment (kayoumido) [inline src/pages/calendar/calendar.ts]: Is this really needed ?

File: src/pages/calendar/calendar.ts [hunk 5]
```diff
@@ -96,5 +104,73 @@ export class CalendarPage {
...
+        const r = calendarList.filter( c => c.name == calendarName)
+        console.log(r)
```
Comment (kayoumido) [inline src/pages/calendar/calendar.ts]: Is this really needed ?

File: src/pages/calendar/calendar.ts (hunk 3 above)
Comment (Grisou13) [reply src/pages/calendar/calendar.ts]: What do you mean?

File: src/pages/calendar/calendar.ts (hunk 5 above)
Comment (Grisou13) [reply src/pages/calendar/calendar.ts]: For debugging on plateforms we haven't yet tested, yes

**Pull request merged**